lib/sysmon/emitter.ex

```diff
@@ -0,0 +1,80 @@
+defmodule Instruments.Sysmon.Emitter do
+  @moduledoc """
+  The Emitter is a simple module that subscribes to the Reporter and will invoke
+  the corresponding handler on the Receiver.
+  """
+
+  use GenServer
+
+  require Logger
+
+  alias Instruments.Sysmon.Reporter
+
+  defstruct [
+    receiver_module: nil
+  ]
+
+  def start_link(opts \\ []) do
+    GenServer.start_link(__MODULE__, opts)
```

Comment:
I think you're missing a `name: __MODULE__` here, judging by the `set_receiver` gen-call (and I don't see it in your supervisor)